packages/pluginutils/test/attachScopes.ts

```diff
@@ -70,3 +70,76 @@ test('supports catch without a parameter', (t) => {
     attachScopes(ast, 'scope');
   });
 });
+
+test('supports ForStatement', (t) => {
+  const ast = (parse(
+    `
+    for (let a = 0; a < 10; a++) {
+      console.log(a);
+      let b = 10;
+    }
+  `,
+    { ecmaVersion: 2020, sourceType: 'module' }
+  ) as unknown) as import('estree').Program;
```

Comment:
Wouldn't it be nicer to just

```js
import {Program, ForStatement,...} from 'estree'
```

once at the top?

Reply:
Also, shouldn't this make "`estree`" and explicit dependency in the `package.json` file?

Reply:
I'm not familiar with typescript so these were copied from:
https://github.com/rollup/plugins/blob/dbececf6c72bb6276f2a80f35a49bbc7a388ace3/packages/pluginutils/src/attachScopes.ts#L64
Should we also move them to the top?

Reply:
Ah I see, I did not notice that one. So `Node` is special in so far as it is also a builtin type so one should be careful to override it, this can cause confusion. Maybe in that case just use a start import instead (works for types just like regular things):

```
import * as estree from 'estree';
```

and then below you use e.g. `estree.Node` or `estree.ForStatement`. Also do not forget to install estree es devDependency, which you should do via `pnpm install -D estree` from the package directory so that the lockfile is updated.